File: task_5/Presenter.java
package task_5;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

import task_5.Interfaces.IChoosingAction;
import task_5.Interfaces.ILogger;
import task_5.Interfaces.IView;

public class Presenter {
    private final IView myView;
    private final IChoosingAction myModel;
    private final ILogger mylogger;
    

    public Presenter(IView myView, IChoosingAction myModel, ILogger mylogger) {
        this.myView = myView;
        this.myModel = myModel;
        this.mylogger = mylogger;
    }

    public void buttonClick() throws IOException {
        double a = myView.getValue("first number");
        String sign = myView.getSign("math char");
        double b = myView.getValue("second number");
        List<Double> operandsList = new ArrayList<>();
        operandsList.add(a);
        operandsList.add(b);
        double result = myModel.chooseModel(sign, operandsList).result();
        String resString = String.format("%f %s %f = %f", a, sign, b, result);
        myView.showResult(resString);
        mylogger.log(Presenter.class.getSimpleName(), "buttonClick", resString);
    }
}

File: task_7/Calc/Presenter.cs
using Calc;


class Presenter
{
    View myView;
    ChooseModel myModel;
    

    
    public Presenter(View myView, ChooseModel myModel) 
    {
        this.myView = myView;
        this.myModel = myModel;
                
    }

    public void ButtonClick()
    {
        double a = myView.GetValue("first number: ");
        string sign = myView.GetSign("math char: ");
        double b = myView.GetValue("second number: ");

        List<double> operandList = new List<double>();
        operandList.Add(a);
        operandList.Add(b);
        
        double result = myModel.ChooseMod(sign, operandList).Result();
       
        myView.ShowResult(result);
        
     
    }
}

